### DISCUSSION TITLE
Help with Nginx Configuration

### DISCUSSION THREAD

**Discussion by BH6AOV** [magicbug/Cloudlog#1027] [General]
**Describe the bug**
After I installed by http://domain/install, database is wrong on the web page：
```text
A Database Error Occurred
Error Number: 1146
Table 'cloudlog_db.options' doesn't exist
SELECT * FROM `options` WHERE `autoload` = 'yes'
Filename: models/Options_model.php
Line Number: 19
```

**To Reproduce**
1.Install LNMP
2.Set database with wiki's guide
3.Install Cloudlog from http://domain/install
4.Open http://domain
5.Saw the error mentioned above.

**Expected behaviour**
normal operation

**Screenshots**
![1234566](https://user-images.githubusercontent.com/30193867/117484326-861f7e00-af99-11eb-8e46-d6db13fb22ce.png)

**Server**
 - OS: Ubuntu 18.04 LTS Server
 - WebServer: nginx/1.14.0 (Ubuntu)
 - Database: mysql  Ver 14.14 Distrib 5.7.33, for Linux (x86_64) using  EditLine wrapper
 - php: PHP 8.0.5 (cli) (built: May  3 2021 11:30:38) ( NTS )

**Additional context**
I tried to manually add an error, but there have been other problems, I hope I can check the problem.I need CloudLog very much.

**Comment by BH6AOV** [comment]
This is a database list after installation.![data](https://user-images.githubusercontent.com/30193867/117484195-57090c80-af99-11eb-89e8-a569a321fd6e.png)

**Comment by magicbug** [comment]
Is this a fresh install? its possibly relating to PHP8 (untested) looks like migrations aren't running.. 

My only suggestion at the moment is use PHP 7.x branch although I know its unpopular choice, will try spend some time getting Cloudlog PHP8 ready soon.

**Comment by BH6AOV** [comment]
> Is this a fresh install? its possibly relating to PHP8 (untested) looks like migrations aren't running..
> 
> My only suggestion at the moment is use PHP 7.x branch although I know its unpopular choice, will try spend some time getting Cloudlog PHP8 ready soon.

Yes, this is a fresh install.
This problem are caused by PHP8, maybe.
I will try remove PHP8 and install PHP 7.3.I will purge all files and fresh reinstall again.If still have this problem, I will feedback directly here.

**Comment by magicbug** [comment]
Might not need to purge the files, just swap PHP version, if you get the same error then do it.

**Comment by BH6AOV** [comment]
@magicbug OK, now it is running, but I can't login while I login the demo account then will give me 404 Not Found.Do you know what happend?
The 404 Page's address is https://domain/index.php/user/login
![404NotFound](https://user-images.githubusercontent.com/30193867/117546474-2db1b480-b05d-11eb-933a-24c4cb370449.png)

**Comment by magicbug** [comment]
Looks like a server config issue, I'd check your logs to see what's up.

**Comment by BH6AOV** [comment]
> Looks like a server config issue, I'd check your logs to see what's up.

uhhh...Which log files do you want?Nginx log Or install log?

**Comment by magicbug** [comment]
Didnt need to see them, just suggested you check the nginx logs to see what your configuration issues are with php.

**Reply by BH6AOV** [reply]
Ok, i will check it.

**Comment by magicbug** [comment]
As this is a issue with server configs rather than Cloudlog I'll change it over to a discussion so you can get help with setting up Nginx.

**Comment by magicbug** [comment]
Edit: Changed Thread title to "Help with Nginx Configuration" to reflect the problem.

**Comment by BH6AOV** [comment]
I got an error log from nginx:
```log
2021/05/10 10:16:12 [error] 6897#6897: *142 open() "/<directory>/index.php/user/login" failed (20: Not a directory), client: <ip>, server: <domain>, request: "POST /index.php/user/login HTTP/1.1", host: "<domain>", referrer: "<domain>"​
```
Could it be said that nginx does not support this opening method?

**Comment by BH6AOV** [comment]
> I got an error log from nginx:
> ```log
> 2021/05/10 10:16:12 [error] 6897#6897: *142 open() "/<directory>/index.php/user/login" failed (20: Not a directory), client: <ip>, server: <domain>, request: "POST /index.php/user/login HTTP/1.1", host: "<domain>", referrer: "<domain>"​
> ```
> Could it be said that nginx does not support this opening method?

I just checked index.php and found that it may be that nginx cannot correctly interpret the file location through index.php. The corresponding file can be seen through domain/apication/views/. If this problem can be fixed, nginx’s  Support should no longer have problems

**Comment by amolea** [comment]
Hello,
I have a ubuntu server runnuing mariadb / php 7.4 and apache 2.4.41. After I copied the git in /var/www/html and trying to access it from http://domain/Cloudlog/install I'm receiving below error:
Not Found
The requested URL was not found on this server.

Apache/2.4.41 (Ubuntu) Server at 192.x.x.x Port 80

**Reply by BH6AOV** [reply]
emmm...Here is Nginx Problem, if you have problem with Apache you can search it to find help or open a new Issues.

**Reply by amolea** [reply]
Thanks! will open new issue.
73's de ON2GXL

**Reply by BH6AOV** [reply]
> Thanks! will open new issue.
> 73's de ON2GXL

73 BH6AOV

**Comment by BH6AOV** [comment]
@magicbug I think I fixed this problem now.I successfully run the program and entered the background by nginx.

**Reply by gvjensen** [reply]
HI BH6AOV - could you share the fix.

I am trying to get config, loaded on https://domainname/Cloudlog to survive the https (it dies).

73's - Glenn

I have php 8 installed, but 7 enabled in Apache2

**Reply by BH6AOV** [reply]
I'm really sorry to hear it and i can't help you because i don't use Apache and don't know how to fix it.

**Reply by resnikov** [reply]
What did you do to fix it?

**Reply by BH6AOV** [reply]
> What did you do to fix it?

I remove php8 and reinstall php7, nginx's php-fpm setting can't use ~ and it work by /.

**Comment by amolea** [comment]
Hi,
To solve it I reinstalled the Nginx using an older version.
Regards,
Alex

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Reply by BH6AOV** [reply]
> Hi, To solve it I reinstalled the Nginx using an older version. Regards, Alex <https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=icon> Virus-free. [www.avast.com](http://www.avast.com) <https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail&utm_term=link> <#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>
> […](#)
> On Fri, Sep 10, 2021 at 9:49 AM resnikov ***@***.***> wrote: What did you do to fix it? — You are receiving this because you commented. Reply to this email directly, view it on GitHub <[#1027 (reply in thread)](https://github.com/magicbug/Cloudlog/discussions/1027#discussioncomment-1305461)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AMM7QLOWLAA3LRMUN4ZGAD3UBGZ73ANCNFSM44PUGBIA> . Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

Old version Nginx can use it? I think i can try it.
